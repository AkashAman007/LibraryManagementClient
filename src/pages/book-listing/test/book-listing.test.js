import React from 'react';
import {render, cleanup, wait } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import  BookListing  from "../book-listing";
import { MemoryRouter } from 'react-router-dom';

afterEach(cleanup);

it("renders book listing", async () => {
    await wait();
    const tree = render(<MemoryRouter><BookListing/></MemoryRouter>)
    expect(tree.asFragment()).toMatchSnapshot();
});

it("library books title", async () => {
    await wait();
    const tree = render(<MemoryRouter><BookListing/></MemoryRouter>)
    const titleElement = tree.getByText("Available Books in Library");
    expect(titleElement).toBeTruthy;
    expect(titleElement).toHaveClass('title');
});

it("book listing has link to my books", async () => {
    await wait();
    const tree = render(<MemoryRouter><BookListing/></MemoryRouter>)
    const linkElement = tree.getByTestId("my-books-link");
    expect(linkElement).toBeTruthy;
    expect(linkElement).toHaveClass('link');
    expect(linkElement).toHaveAttribute('href');
    expect(linkElement.getAttribute('href')).toEqual('/MyBooks');
});

it("when no books in library", async () => {
    await wait();
    const tree = render(<MemoryRouter><BookListing/></MemoryRouter>)
    const messageElement = tree.getByText("No books in library");
    expect(messageElement).toBeTruthy;
});
