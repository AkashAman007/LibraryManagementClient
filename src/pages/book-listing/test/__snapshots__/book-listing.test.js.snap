// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders book listing 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="title"
    >
      Available Books in Library
      <a
        class="float-right link"
        data-testid="my-books-link"
        href="/MyBooks"
      >
        See Borrowed Books
      </a>
    </h1>
    <div
      class="book-grid"
    >
      <div>
        No books in library
      </div>
    </div>
  </div>
</DocumentFragment>
`;
